Modules/mmapmodule.c

```diff
@@ -1097,8 +1268,13 @@ mmap_ass_subscript(mmap_object *self, PyObject *item, PyObject *value)
             return -1;
         }
         CHECK_VALID(-1);
-        self->data[i] = (char) v;
-        return 0;
+
+        if (safe_byte_copy(self->data + i, (char *) &v) < 0) {
```

Comment:
```suggestion
        char v_char = (char) v;
        if (safe_byte_copy(self->data + i, &v_char) < 0) {
```

There's a subtle difference between the casts, so safest to keep casting the value to `char` and then reference the address.